src/components/app/router.tsx

```diff
@@ -0,0 +1,106 @@
+import { Component, FunctionalComponent, h, Prop, State } from '@stencil/core';
+import { URLPattern } from 'urlpattern-polyfill';
+
+type MatchedComponent = {
+    name: string;
+    props?: Record<string, string | number>;
+};
+
+type RoutedComponent = {
+    path: string;
+    name: string;
+    props?: Record<string, string | number>;
+};
+
+const ComponentTemplate: FunctionalComponent<MatchedComponent> = (
+    props,
+    children,
+) => {
+    const Name = props.name;
+
+    return <Name {...props.props}>{children}</Name>;
+};
+
+@Component({
+    tag: 'kompendium-router',
+    shadow: true,
+})
+export class Router {
+    @Prop()
+    public routes: RoutedComponent[];
+
+    @State()
+    private matchedComponent: MatchedComponent | undefined;
+
+    private hashChangeHandler = () => {
```

Comment:
I got the routing to work, but I'm currently stuck at the code in `markdown.ts` not working correctly. It just returns `undefined`. I've added some logging, to help with debugging the problem, but not gotten any further.